skimage/feature/peak.py

```diff
@@ -133,60 +176,41 @@ def peak_local_max(image, min_distance=1, threshold_abs=None,
             labels[mask] = 1 + rank_order(labels[mask])[0].astype(labels.dtype)
         labels = labels.astype(np.int32)
 
-        # New values for new ordering
-        label_values = np.unique(labels)
-        for label in label_values[label_values != 0]:
-            maskim = (labels == label)
-            out += peak_local_max(image * maskim, min_distance=min_distance,
-                                  threshold_abs=threshold_abs,
-                                  threshold_rel=threshold_rel,
-                                  exclude_border=exclude_border,
-                                  indices=False, num_peaks=num_peaks_per_label,
-                                  footprint=footprint, labels=None)
-
-        # Select highest intensities (num_peaks)
-        coordinates = _get_high_intensity_peaks(image, out, num_peaks)
+        if exclude_border:
+            # create a mask for the non-exclude region
+            inner_mask = _exclude_border(np.ones_like(labels, dtype=bool),
+                                         footprint, exclude_border)
+
+        for i, obj in enumerate(ndi.find_objects(labels)):
+            label = i + 1
```

Comment:
This variable seems to be unused... (Am i correct?)

Reply:
The variable label is used in the next line:

`img=image[obj]*(labels[obj]==label)`